helix-term/src/ui/picker.rs

```diff
@@ -165,34 +166,48 @@ impl<T: Item> FilePicker<T> {
 
 impl<T: Item + 'static> Component for FilePicker<T> {
     fn render(&mut self, area: Rect, surface: &mut Surface, cx: &mut Context) {
-        // +---------+ +---------+
-        // |prompt   | |preview  |
-        // +---------+ |         |
-        // |picker   | |         |
-        // |         | |         |
-        // +---------+ +---------+
-
-        let render_preview = self.picker.show_preview && area.width > MIN_AREA_WIDTH_FOR_PREVIEW;
+        use helix_view::editor::FilePickerLayoutDirection::*;
+
+        let layout_direction = cx.editor.config().borrow().file_picker.layout_direction;
+        // Horizontal layout:         Vertical layout:
+        // +---------+ +---------+    +---------------------+
+        // |prompt   | |preview  |    | prompt              |
+        // +---------+ |         |    +---------------------+
+        // |         | |         |    | picker              |
+        // |picker   | |         |    +---------------------+
+        // |         | |         |    +---------------------+
+        // |         | |         |    | preview             |
+        // +---------+ +---------+    +---------------------+
```

Comment:
I moved the layout code into a function, returning a `(Rect, Option<Rect>)` for the picker/preview area.

Maybe I should return a proper struct instead.

I also made sure the `fn cursor` passes the `picker_area` as defined in the layout function down to self.picker now.

Testing my vertical layout with short terminal windows, I see the picker area shrink down to 2 lines with no space for the input line and any choices.

It behaves much more sensibly in horizontal mode. Might that be due to the `request_size` method? That seemed strange to me before -- and still is strange now:-)

Reply:
Layout seems sensible, even in small sizes now with all the changes to make the weight of the widgets configurable. This also gets rid of a magic number you did not like earlier:-)

Reply:
Nice :+1: The layouts should be sensible no matter the terminal size in my opinion.

I'm not strictly against magic numbers, what often gets lost (with time or for new contributors) is the "why" of the magic numbers. Were they chosen because the author thought it looked good or is there some more fundamental background behind the choice? This can clarified in multiple ways; in code, in comments or in the commit message, etc.